### PR TITLE
Igor lig 1788 smog add examples

### PR DESCRIPTION
**Pull request by IgorSusmelj**
## Changes

- add SMoG pytorch example
- add SMoG to readme

Changes by @philippmwirth:
- Refactor SMoG to use parameters instead of tensors
- Clean up benchmarks
- Add SMoG pytorch-lightning example

Note: I did not write a multi-gpu example because I don't think our implementation can handle it atm.

### CODE REVIEW

**Comment by IgorSusmelj** [comment]
Handed over PR to @philippmwirth

File: docs/source/getting_started/benchmarks/cifar10_benchmark.py
Comment (IgorSusmelj) [inline docs/source/getting_started/benchmarks/cifar10_benchmark.py:847]: That's still a leftover of testing?

File: lightly/models/modules/heads.py
Comment (IgorSusmelj) [inline lightly/models/modules/heads.py:275]: Sure we want to keep this in?

File: lightly/models/modules/heads.py
Comment (IgorSusmelj) [inline lightly/models/modules/heads.py:281]: Same here

**Review by IgorSusmelj — COMMENTED** [review]
Haha, I neither can't approve nor request changes as I'm the author^^

I left a few comments. We're still missing the other two examples (pytorch lightning and distributed)

**Comment by philippmwirth** [comment]
> Haha, I neither can't approve nor request changes as I'm the author^^
> 
> I left a few comments. We're still missing the other two examples (pytorch lightning and distributed)

Hm somehow I didn't push the pytorch lightning example. I decided that it doesn't make sense to have a distributed example if `SMoG` is not implemented for distributed setups yet.

**Comment by philippmwirth** [comment]
> > I decided that it doesn't make sense to have a distributed example if SMoG is not implemented for distributed setups yet.
> 
> Ok, then I'd make a note to page (if users click on the smog distributed button) that this is not supported yet and will follow soon. And please don't forget to create the follow up issue so we don't forget about this.

There is no more distributed button.

File: docs/source/getting_started/benchmarks/cifar10_benchmark.py
Comment (philippmwirth) [reply docs/source/getting_started/benchmarks/cifar10_benchmark.py:847]: Yes, I'll remove it before merging 👍

File: docs/source/getting_started/benchmarks/cifar10_benchmark.py
Comment (guarin) [inline docs/source/getting_started/benchmarks/cifar10_benchmark.py:856]: Uncomment

File: docs/source/getting_started/benchmarks/imagenette_benchmark.py
Comment (guarin) [inline docs/source/getting_started/benchmarks/imagenette_benchmark.py:873]: Why is this no longer 300 * batch_size ?

**Review by guarin — COMMENTED** [review]
Code and examples look great! I am running the examples and will approve once they ran through.

**Review by guarin — APPROVED** [review]
Examples ran through and loss decreases 💯

File: docs/source/getting_started/benchmarks/imagenette_benchmark.py
Comment (philippmwirth) [reply docs/source/getting_started/benchmarks/imagenette_benchmark.py:873]: Because `300 * batch_size >> n_samples`

**Pull request merged**
